tools/cli/src/cli/session.rs

```diff
@@ -641,19 +645,24 @@ impl LogoutOpt {
                 }
             };
 
-            let jws_verifier = if let Some(pub_jwk) = jwsc.get_jwk_pubkey() {
-                match JwsEs256Verifier::try_from(pub_jwk) {
-                    Ok(verifier) => verifier,
-                    Err(err) => {
-                        error!(?err, "Unable to configure jws verifier");
-                        std::process::exit(1);
-                    }
-                }
-            } else {
-                error!("Unable to access token public key");
+            let Some(key_id) = jwsc.kid() else {
+                error!("Invalid token, missing KeyID");
```

Comment:
Since this is still parsing the local token, if it's invalid why not just log a debug message and remove it from the local store? (same with the rest of this)

Reply:
Yeah, I ummed and ahhed a lot about this. The problem is that we need the *spn* to actually ID the token out the store. The spn is collected down in to_client(), and by this part of the process we *must* have done a successful to_client() which means:

* The token *already* has to have passed signature checks for to_client to have worked: https://github.com/kanidm/kanidm/blob/master/tools/cli/src/cli/common.rs#L213
* If logout() succeeded that means the token was *also* validated.

So by this point, unless something bonkers happens, then the validation will always pass.

I did consider using ReleaseNoVerify here, but I think the above meant it was "safe" by that point. 

Really, this is a bit of a tough one layering wise, because we have to handle token state in the client *and* in the cli itself due to the token store. Tech Debt, yay. 

Anywy, I think an addition that could be wise here is removing https://github.com/kanidm/kanidm/blob/master/tools/cli/src/opt/kanidm.rs#L785 hidden from the local_only flag, allowing the token to be removed *always* bypassing all these other checks.

Reply:
> So by this point, unless something bonkers happens, then the validation will always pass.

so why error out and leave a broken token in the local cache at that point? just break out of the parsing, remove it from the file and carry on

Reply:
Because we haven't been able to bubble up the spn that we need to remove the token.

Reply:
This seems like a logical loop. 

1. We got the token from the local token store.
2. We tried to parse it
3. That failed 
4. Now we can't remove it because we can't parse it?

Reply:
Which is exactly why I un-hide the local only removal flow. That allows breaking out of the loop.

Reply:
But it's a broken token, why not just clean the token store and tell the user what happened, instead of `exit(1)`ing with an error message and leaving them with the broken thing still in the cache? 

Fair enough don't try and send it to the server, but it shouldn't be in the token store at the end of the interaction if it's broken - once you've gotten past line 636 (`client.get_token()`) there's *something* found that can be dealt with, so if validation doesn't work then just clean it up.

Reply:
That's a good point. I think we need to do that in the try_to_client() call though because that affects every possible workflow / interaction.

Reply:
That'd work, too 😄

Reply:
Reading through this, the logic to add that here is going to be a nightmare actually. I think having unhidden the ability to force remove the token locally is going to have to suffice for now. Also note that kanidm session cleanup catches them too.

Reply:
Actually there is a way, spoke too soon.

Reply:
Actually, there isn't.